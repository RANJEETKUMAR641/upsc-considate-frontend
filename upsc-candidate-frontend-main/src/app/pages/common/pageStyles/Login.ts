import styled from 'styled-components'
interface Props {}

export const LoginStyle = styled('div')<Props>`
  .main-card {
    border: 0px !important;
    padding: 0px !important;
  }
  .loginSection {
    padding: 0px !important;
  }
  .loginNameBox {
    width: 100%;
    text-align: center;
    padding-top: 0.7rem;
  }
  .card-body {
    margin-top: 12px;
    margin-bottom: 12px;
  }
  .loginLogo {
    display: block;
  }
  .loginLogo img {
    width: 70px;
  }
  .loginNameBox h1 {
    font-size: 1.2rem;
    font-weight: 700;
    line-height: 1.2rem;
  }
  .loginNameBox h1 span {
    font-size: 0.6rem;
    display: block;
  }
  .logInLogoBox img {
    width: 50px !important;
  }
  .loginBox {
    padding: 0.4rem 0.4rem;
    max-width: 400px;
    margin: auto;
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    left: 15px;
    right: 15px;
    z-index: 10;
  }
  .loginSection {
    border: 1px solid #dee2e6;
    padding: 1rem 2rem;
    background: #fff;
  }

  .MuiPaper-root {
    border-radius: 0;
    border: none;
  }

  @media only screen and (max-width: 767.98px) {
    .loginBox {
      position: static;
      transform: translateY(0%);
    }
    .loginSection {
      padding: 1rem 1rem;
    }
    .loginLogo {
      display: none;
    }
    .candidatepage {
      padding: 0;
    }
    .mobileHide {
      display: none;
    }
  }

  /* Media query for viewport height less than or equal to 500px */
  @media (max-height: 500px) {
    .loginBox {
      top: 100px;
      transform: none;
    }
  }
`
