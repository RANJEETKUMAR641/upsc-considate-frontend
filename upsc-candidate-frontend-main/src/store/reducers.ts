/* eslint-disable */

import { combineReducers } from 'redux'
import reducers from 'app/components/reducers'
import { connectRouter } from 'connected-react-router'

import history from 'utils/history'

import { InjectedReducersType } from 'utils/types/injector-typings'

export default function createReducer(
  injectedReducers: InjectedReducersType = {},
) {
  return combineReducers({
    ...injectedReducers,
    ...reducers,
    router: connectRouter(history),
  })
}
