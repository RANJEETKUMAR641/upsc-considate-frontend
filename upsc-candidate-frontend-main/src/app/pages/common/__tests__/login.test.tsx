import { render, screen, fireEvent, logRoles } from '@testing-library/react'

import userEvent from '@testing-library/user-event'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { configureAppStore } from 'store/configureStore'

import Login from '../login'
import { Provider } from 'react-redux'

jest.mock('axios')

jest.mock('')
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    }
  },
}))

jest.mock('utils/requestPayload', () => {
  return {
    data: { token: 'dummy_token' },
  }
})

const mockedAlert = jest.fn()
global.alert = mockedAlert

// describe('Test cases for login user', () => {
//   const queryClient = new QueryClient()
//   const store = configureAppStore()

//   it('should login new user', async () => {
//     const { result, waitFor } = renderHook(() => useGetLogin(), {
//       wrapper: wrapper,
//     })
//     nock('http://localhost:3000', {
//       reqheaders: {
//         'app-id': () => true,
//       },
//     })
//       .post('/api/user/login')
//       .reply(200, {})

//     act(() => {
//       result.current.mutate({
//         email: 'dummy@dummy.com',
//         password: 'asdA@123',
//       } as any)
//     })
//     await waitFor(() => {
//       return result.current.isLoading
//     })
//     expect(result.current.isLoading).toBe(true)
//   })
// })

describe('Test cases for login page snapshot', () => {
  const queryClient = new QueryClient()
  const store = configureAppStore()
  it('should match snapshot', () => {
    const loadingIndicator = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Login />
        </QueryClientProvider>
      </Provider>,
    )
    expect(loadingIndicator.container.firstChild).toMatchSnapshot()
  })
})

describe('Test cases for login form elements (email,password,login are present', () => {
  it('should element present in document', () => {
    const queryClient = new QueryClient()
    const store = configureAppStore()

    const { getByLabelText, getByText, getByRole } = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Login />
        </QueryClientProvider>
      </Provider>,
    )

    expect(getByLabelText('Email')).toBeInTheDocument()
    expect(getByLabelText('Password')).toBeInTheDocument()
    expect(getByText('Login')).toBeInTheDocument()
    expect(
      getByRole('link', { name: 'One Time Registration (OTR)' }),
    ).toBeInTheDocument()
  })
})

describe('Test cases for displays login form validation', () => {
  it('displays validation errors for invalid input', async () => {
    const queryClient = new QueryClient()
    const store = configureAppStore()

    const { getByText } = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Login />
        </QueryClientProvider>
      </Provider>,
    )

    userEvent.click(getByText('Login'))
    const email = await screen.findByText('Email is required')
    const password = await screen.findByText('Password is required')

    expect(email).toBeInTheDocument()
    expect(password).toBeInTheDocument()
  })
})

describe('Test cases for link', () => {
  it('test for link of One Time Registration', async () => {
    const queryClient = new QueryClient()
    const store = configureAppStore()

    const { getByRole } = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Login />
        </QueryClientProvider>
      </Provider>,
    )

    expect(
      getByRole('link', { name: 'One Time Registration (OTR)' }),
    ).toHaveAttribute('href', '/')
  })
})

//fail prevent default

// describe('Test cases recover password link', () => {
//   it('test for link of recover password', async () => {
//     const queryClient = new QueryClient()
//     const store = configureAppStore()

//     const { getByRole } = render(
//       <Provider store={store}>
//         <QueryClientProvider client={queryClient}>
//           <Login />
//         </QueryClientProvider>
//       </Provider>,
//     )
//     const recoverPassword = getByRole('link', { name: 'Recover Password' }));
//     fireEvent.click(getByRole('link', { name: 'Recover Password' }));
//     expect().toHaveAgetByRole('link', { name: 'Recover Password' })attribute('href', '/#')
//   })
// })

describe('Test cases for mock email', () => {
  it('email validation and displays it', async () => {
    const queryClient = new QueryClient()
    const store = configureAppStore()

    const { getByRole } = render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Login />
        </QueryClientProvider>
      </Provider>,
    )

    const inputFirst = getByRole('textbox', { name: /email/i })
    const submitButton = getByRole('button', { name: /Login/i })
    userEvent.type(inputFirst, 'abcgmail.com')
    userEvent.tab()
    userEvent.click(submitButton)
    expect(inputFirst).toBeInTheDocument()
    expect(submitButton).toBeInTheDocument()
  })
})

describe('Test cases for Email validation error', () => {
  it('displays  errors for invalid email', async () => {
    const queryClient = new QueryClient()
    const store = configureAppStore()

    render(
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <Login />
        </QueryClientProvider>
      </Provider>,
    )
    logRoles(screen.getByTestId('myForm'))
    const emailInput = screen.getByTestId('email')
    fireEvent.change(emailInput, { target: { value: 'invalid-email' } })
    const submitButton = screen.getByText('Login')
    fireEvent.click(submitButton)
    const validation = await screen.findByText('Enter a valid email')
    expect(validation).toBeInTheDocument()
  })
})

//prevent default

// describe('Test cases for focus', () => {
//   test('test focus and Blur', async () => {
//     const queryClient = new QueryClient()
//     const store = configureAppStore()

//     const { getByLabelText } = render(
//       <Provider store={store}>
//         <QueryClientProvider client={queryClient}>
//           <Login />
//         </QueryClientProvider>
//       </Provider>,
//     )

//     const input1 = getByLabelText('Email')
//     const input2 = getByLabelText('Password')

//     input1.focus()
//     fireEvent.change(input1, { target: { value: '' } })
//     userEvent.tab()

//     expect(input2).toHaveFocus()
//   })
// })

// describe('Test cases for password validation error', () => {
// it('display error for short password', async() => {
//   const queryClient = new QueryClient()
//     const store = configureAppStore()
//     const { getByText} = render(
//       <Provider store={store}>
//         <QueryClientProvider client={queryClient}>
//           <Login />
//         </QueryClientProvider>
//       </Provider>,
//     )
//   const passwordInput = screen.getByTestId("Password");
//  fireEvent.change(passwordInput,{target:{value: 'abc'}});
//   const submitButton = screen.getByText('Login');
//   fireEvent.click(submitButton);
// const validation = await screen.findByText('Password should be at least 8 characters long')
//   expect(validation).toBeInTheDocument();
// });
// });

// case fail
// describe('not show any error message', () => {
//   it('should not show any error message when the component is loaded', async() => {
//     const queryClient = new QueryClient()
//       const store = configureAppStore()

//       const { getByText} = render(
//         <Provider store={store}>
//           <QueryClientProvider client={queryClient}>
//             <Login />
//           </QueryClientProvider>
//         </Provider>,
//       )
//   const alertElement = screen.findByRole("alert");
//   expect(alertElement).not.toBeInTheDocument();
// });
// });

//(Case Fail)
// describe('Test cases for mock password', () => {
//   it('invalid password validation and displays it', async () => {
//     const queryClient = new QueryClient()
//     const store = configureAppStore()

//     const { getByRole, getByLabelText } = render(
//       <Provider store={store}>
//         <QueryClientProvider client={queryClient}>
//           <Login />
//         </QueryClientProvider>
//       </Provider>,
//     )

//     fireEvent.input(getByRole('textbox', { name: /email/i }), {
//       target: {
//         value: 'test',
//       },
//     })

//     fireEvent.input(getByLabelText('password'), {
//       target: {
//         value: 'password',
//       },
//     })

//     fireEvent.submit(getByRole('button'))

//     expect(await screen.findAllByRole('alert')).toHaveLength(1)
//     expect(Login).not.toBeCalled()
//     expect(getByRole('textbox', { name: /email/i })).toHaveValue('test')
//     expect(getByLabelText('password')).toHaveValue('password')
//   })
// })
