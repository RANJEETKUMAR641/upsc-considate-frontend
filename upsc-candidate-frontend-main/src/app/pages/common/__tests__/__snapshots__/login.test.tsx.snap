// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for login page snapshot should match snapshot 1`] = `
<div
  class="sc-hKMtZM ePzAKj"
>
  <div
    class="fixed-header bg-white"
    style="height: 60px; position: relative;"
  >
    <div
      class="fixed-header"
    >
      <div>
        <div
          aria-label="Site Header"
          class="app-header header-shadow appear appear-active"
          component="div"
        >
          <div
            class="app-header__logo"
          >
            <div
              class="sc-bczRLJ kxxrZ logoTxt"
            >
              <img
                alt="Logo"
                src="IMAGE_MOCK"
              />
              <h1
                class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
              >
                संघ लोक सेवा आयोग
                <p>
                  UNION PUBLIC SERVICE COMMISSION
                </p>
              </h1>
            </div>
          </div>
          <div
            class="app-header__mobile-menu"
          >
            <div
              role="button"
              tabindex="0"
            >
              <button
                class="jsx-3928250682 BurgerSlider Burger"
              >
                <div
                  class="jsx-3928250682 BurgerBox"
                >
                  <div
                    class="jsx-3928250682 BurgerInner"
                  />
                </div>
              </button>
            </div>
          </div>
          <div
            class="app-header__menu"
          >
            <span
              role="button"
              tabindex="0"
            >
              <button
                class="btn-icon btn-icon-only btn btn-dark btn-sm"
                type="button"
              >
                <div
                  class="btn-icon-wrapper"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-ellipsis-v "
                    data-icon="ellipsis-v"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="vertical-align: 0rem;"
                    viewBox="0 0 192 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </button>
            </span>
          </div>
          <div
            class="app-header__content"
          >
            <div
              class="app-header-right"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="loginBox"
  >
    <div
      class="loginSection card"
    >
      <div
        class="card-body"
      >
        <div
          class="loginLogo"
        >
          <div
            class="logInLogoBox"
          >
            <img
              alt="Logo"
              aria-hidden="true"
              class="img-fluid mx-auto d-block"
              src="IMAGE_MOCK"
            />
          </div>
          <div
            class="loginNameBox"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
            >
              संघ लोक सेवा आयोग
              <span>
                UNION PUBLIC SERVICE COMMISSION
              </span>
            </h1>
          </div>
        </div>
        <div
          class="divider"
        />
        <div
          class="myform"
        >
          <div
            class="sc-eCYdqJ eklDXt"
          >
            <form>
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <div
                    class="mb-3"
                    data-testid="myForm"
                  >
                    <label
                      class="my-label form-label"
                      for="exampleEmail"
                      tabindex="5"
                    >
                      Email
                    </label>
                    <input
                      aria-describedby="email-error"
                      aria-label="Email"
                      aria-required="true"
                      autocomplete="off"
                      class="input-sm form-control"
                      data-testid="email"
                      id="email"
                      name="email"
                      placeholder="Email here..."
                      tabindex="6"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-md-12"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="my-label form-label"
                      for="examplePassword"
                      tabindex="7"
                    >
                      Password
                    </label>
                    <input
                      aria-describedby="password-error"
                      aria-label="Password"
                      aria-required="true"
                      autocomplete="off"
                      class="input-sm form-control"
                      id="password"
                      name="password"
                      placeholder="Password here..."
                      tabindex="7"
                      type="password"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="col-md-12"
                >
                  <div
                    class="mb-3"
                  >
                    <label
                      class="my-label form-label"
                      for="exampleCaptcha"
                      tabindex="7"
                    >
                      Captcha
                    </label>
                    <div
                      class="d-flex"
                    >
                      <div
                        aria-label="captcha"
                        class=""
                        data-mui-internal-clone-element="true"
                      />
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                        style="text-align: left;"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="material-symbols-outlined text-info"
                        >
                          <svg
                            aria-label="refresh"
                            data-mui-internal-clone-element="true"
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            style="font-size: 30px; display: inline-block; margin-top: 10px; margin-bottom: 10px;"
                            viewBox="0 0 512 512"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M256 388c-72.597 0-132-59.405-132-132 0-72.601 59.403-132 132-132 36.3 0 69.299 15.4 92.406 39.601L278 234h154V80l-51.698 51.702C348.406 99.798 304.406 80 256 80c-96.797 0-176 79.203-176 176s78.094 176 176 176c81.045 0 148.287-54.134 169.401-128H378.85c-18.745 49.561-67.138 84-122.85 84z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <input
                      aria-describedby="captcha-error"
                      aria-label="Captcha"
                      aria-required="true"
                      autocomplete="off"
                      class="input-sm form-control"
                      id="captcha"
                      name="captcha"
                      placeholder="Captcha here..."
                      tabindex="7"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="divider"
              />
              <div
                class="mb-4"
              >
                <button
                  class="btn btn-info button-sm"
                  style="float: right;"
                  tabindex="8"
                  type="submit"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall me-2 css-ptiqhd-MuiSvgIcon-root"
                    data-testid="LoginOutlinedIcon"
                    focusable="false"
                    style="margin-top: -2px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
                    />
                  </svg>
                  Login
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
